### PR TITLE
Fix incorrect value from getTotalTokens when fees are cached

### PR DESCRIPTION
**Pull request by zathras-crypto**
When a property was not created via a fixed issuance, we use a method of iterating over the tally map to calculate the total number of tokens.

However after fees are activated, some tokens may not be in the tally, instead being held in the fee cache.

`getTotalTokens()` must be made aware of the fee cache and needs to include cached fees in the total count of tokens.

This change is consensus breaking.

### CODE REVIEW

**Comment by zathras-crypto** [comment]
I didn't quite know what to do about this bud, so I've put it in as a separate PR.

**Comment by dexX7** [comment]
Hmm.. do we need to count cached amounts at all?

**Comment by zathras-crypto** [comment]
> Hmm.. do we need to count cached amounts at all?

If we want to provide accurate data :)

Let's take a couple examples:
1) We trade some Tether and Tether gets moved to the fee cache.  When we now call something like `getProperty` on the RPC layer the information we provide about the number of tokens would be incorrect.
2) I create a property via crowdsale.  While the crowdsale is active we trade some of the property meaning some of the tokens are moved to the fee cache.  Since `getTotalTokens()` is now inaccurate the crowdsale helpers can make a mess of the math, including creating more than MAX_INT64 tokens.

So to answer your question, yes I really do believe we need to count cached amounts - `getTotalTokens()` us a fairly critical function and I don't really consider it feasible to leave it returning incorrect data.

TL:DR; if we activate fees as is there are scenarios where we could break things and I'm a bit worried about that.

**Comment by zathras-crypto** [comment]
Note - I know this was my mistake when I wrote the fees code, and I know no-one is going to like needing a fix before we can activate fees, but I just didn't see the problem before now :(

**Comment by dexX7** [comment]
> TL:DR; if we activate fees as is there are scenarios where we could break things and I'm a bit worried about that.

Thanks for laying out examples. I agree, we really shouldn't activate before this is fixed. :)

Hmm.. so this makes 0.2 consensus breaking. Do you think we should also publish an updated 0.10 based version? I think we might, if we release them simultaneously.

**Comment by zathras-crypto** [comment]
> Hmm.. so this makes 0.2 consensus breaking. Do you think we should also publish an updated 0.10 based version? I think we might, if we release them simultaneously.

Yeah, that's what I really didn't like about the change.  I really liked the idea of having 0.2 be a release that didn't change any rules.

Perhaps we could do a `0.0.11.3-rel` with the fix and that becomes the stable version for now.  Then when 0.2 comes it wouldn't contain any new rules.  If we did another 0.10 base release there is the option to backport the perf improvements too.  I'm not really sure though.

I'm also going to take another look at fees to see if there is anything else I can find.

**Comment by dexX7** [comment]
> Perhaps we could do a 0.0.11.3-rel with the fix and that becomes the stable version for now.

Sounds fine to me. We could then release it with 0.2, and provide a choice for users.

> If we did another 0.10 base release there is the option to backport the perf improvements too. I'm not really sure though.

Hmm.. we should encourage people to use 0.2 anyway, so I don't think it's worth porting, but I wouldn't mind either. :)
